src/state/RechartsStoreProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode, useRef } from 'react';
+import { Provider } from 'react-redux';
+import { createRechartsStore, RechartsRootState } from './store';
+
+type RechartsStoreProviderProps = {
+  children: ReactNode;
+  preloadedState?: Partial<RechartsRootState>;
+  reduxStoreName?: string;
+};
+
+export function RechartsStoreProvider({ preloadedState, children, reduxStoreName }: RechartsStoreProviderProps) {
+  const storeRef = useRef(null);
```

Comment:
That's right but that's considering a singleton app. In recharts, we want to be able to support multiple independent charts on one page.

The alternative is to have everything in the store keyed by the chart id. Which makes working with everything a little bit more painful because we need the chart id everywhere. And it makes synchronising charts easier.

Reply:
This is very nice reasoning. Could you please add that as a comment to the code? Otherwise I am afraid in the distant future someone will just be like: Oh, why is this a ref, lets fix it.

Reply:
I added a comment. I tried to write a unit test to demonstrate that too.